python/lsst/pipe/tasks/dataFrameActions/_actions.py

```diff
@@ -0,0 +1,135 @@
+from ._baseDataFrameActions import DataFrameAction
+from ._evalColumnExpression import makeColumnExpressionAction
+
+from lsst.pex.config import Field
+
+
+class SingleColumnAction(DataFrameAction):
+    column = Field(doc="Column to load for this action", dtype=str, optional=False)
+
+    @property
+    def columns(self) -> Iterable[str]:
+        return (self.column, )
+
+    def __call__(self, df, **kwargs):
+        return df[self.column]
+
+
+class MultiColumnAction(DataFrameAction):
+    actions = ConfigurableActionStructField(doc="Configurable actions to use in a joint action")
+
+    @property
+    def columns(self) -> Iterable[str]:
+        yield from (column for action in self.actions for column in action.columns)
+
+
+class CoordColumn(SingleColumnAction):
+    inRadians = Field(doc="Return the column in radians if true", default=True, dtype=bool)
+
+    def __call__(self, df):
+        col = super().__call__(df)
+        return col * 180 / np.pi if self.inRadians else col
+
+
+class MagColumn(SingleColumnAction):
+    coadd_zeropoint = Field(doc="Magnitude zero point", dtype=float, default=27)
+
+    def __call__(self, df: pd.DataFrame, **kwargs):
+        if not (fluxMag0 := kwargs.get('fluxMag0')):
+            fluxMag0 = 1/np.power(10, -0.4*self.coadd_zeropoint)
+
+        with np.warnings.catch_warnings():
+            np.warnings.filterwarnings('ignore', r'invalid value encountered')
+            np.warnings.filterwarnings('ignore', r'divide by zero')
+            return -2.5 * np.log10(df[self.column] / fluxMag0)
+
+
+class NanoJansky(SingleColumnAction):
+    ab_flux_scale = Field(doc="Scaling of ab flux", dtype=float, default=(0*units.ABmag).to_value(units.nJy))
+    coadd_zeropoint = Field(doc="Magnitude zero point", dtype=float, default=27)
+
+    def __call__(self, df, **kwargs):
+        dataNumber = super().__call__(df, **kwargs)
+        if not (fluxMag0 := kwargs.get('fluxMag0')):
+            fluxMag0 = 1/np.power(10, -0.4*self.coadd_zeropoint)
+        return self.ab_flux_scale * dataNumber / fluxMag0
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.cache = True  # cache this action for future calls
+
+
+class NanoJanskyErr(SingleColumnAction):
+    flux_mag_err = Field(doc="Error in the magnitude zeropoint", dtype=float, default=0)
+    flux_action = ConfigurableActionField(doc="Action to use if flux is not provided to the call method",
+                                          default=NanoJansky, dtype=DataFrameAction)
+
+    @property
+    def columns(self):
+        yield from zip((self.column,), self.flux_action.columns)
+
+    def __call__(self, df, flux_column=None, flux_mag_err=None, **kwargs):
+        if flux_column is None:
+            flux_column = self.flux_action(df, **kwargs)
+        if flux_mag_err is None:
+            flux_mag_err = self.flux_mag_err
+
+
+docs = """This is a `DataFrameAction` that is designed to add two columns
+together and return the result.
+"""
```

Comment:
Indent is off

Reply:
This is a module level variable (that maybe should have a _ in it) and is declared with a tipple quoted string, so we want to keep the internal formatting as is.